crates/oxide/src/extractor/pre_processors/haml.rs

```diff
@@ -173,10 +421,20 @@ mod tests {
 
     // https://github.com/tailwindlabs/tailwindcss/pull/17051#issuecomment-2711181352
     #[test]
-    fn test_haml_full_file() {
-        let processed = Haml.process(include_bytes!("./test-fixtures/haml/src-1.haml"));
+    fn test_haml_full_file_17051() {
+        let processed = Haml.process(include_bytes!("./test-fixtures/haml/src-17051.haml"));
+        let actual = std::str::from_utf8(&processed).unwrap();
+        let expected = include_str!("./test-fixtures/haml/dst-17051.haml");
+
+        assert_eq!(actual, expected);
```

Comment:
I find this a bit confusing to review. Looking at the snapshot alone I am really not sure if the right classes are extract. Should we run the extractor here and assert on the actual candidates being emitted?